src/server/services/airtable.ts

```diff
@@ -0,0 +1,506 @@
+      const oldDemands = !!gestionnaire.get('Relance');
+      const active = !!gestionnaire.get('Actif');
+      const allTags = [...new Set([...tags, ...tagsFromAPI])];
+      const user = users.find((user) => user.email.toLowerCase() === email.toLowerCase());
+
+      if (!user) {
+        logDry(`    🆕 Create user for ${email} with ${allTags.join(',')}.`);
+        const data = {
+          email: email.toLowerCase(),
+          password: await bcrypt.hash(Math.random().toString(36).slice(2, 10), salt),
+          gestionnaires: allTags,
+          receive_new_demands: newDemands,
+          receive_old_demands: oldDemands,
+          active,
+          role: USER_ROLE.GESTIONNAIRE,
+        };
+
+        if (!DRY_RUN && process.env.NODE_ENV !== 'production') {
+          logger.error('As creation of users in database will send email to users, you need to set NODE_ENV=production');
+          return;
+        }
+
+        if (!DRY_RUN && process.env.NODE_ENV === 'production') {
+          try {
+            await db('users').insert(data);
+            await base(Airtable.GESTIONNAIRES).update(
+              gestionnaire.id,
+              {
+                'Créé en base': new Date().toISOString(),
+              },
+              { typecast: true }
+            );
+          } catch (e) {
+            logger.error(`Could not create ${email} in database`);
+            logger.error(e);
+          }
+        }
+
+        logDry(`    📩 Sending inscription email to ${email}`);
+        if (!DRY_RUN && process.env.NODE_ENV === 'production') await sendInscriptionEmail(email);
+        stats.totalCreated++;
+        return;
+      }
+
+      if (!active && user.active) {
+        logDry(`    ❌ Deactivate user ${email}`);
+        if (!DRY_RUN) {
+          try {
+            await db('users').update('active', false).where('id', user.id);
+          } catch (e) {
+            logger.error(`Could not create ${email} in database`);
+            logger.error(e);
+          }
+        }
+        stats.totalDeactivated++;
+        return;
+      }
+      const { unchanged } = diff(user.gestionnaires || [], allTags);
+
+      if (
+        unchanged.length === allTags.length &&
+        user.receive_new_demands === newDemands &&
+        user.receive_old_demands === oldDemands &&
+        user.active === active
+      ) {
+        logDry(`    💤 No changes for ${email}`);
+        return;
+      }
+
+      if (!DRY_RUN) {
+        logDry(`    ✅ Update gestionnaires for ${email}`);
+
+        try {
+          await db('users')
+            .update({
+              gestionnaires: allTags,
+              receive_new_demands: newDemands,
+              receive_old_demands: oldDemands,
+              active,
+            })
+            .where('id', user.id);
+        } catch (e) {
+          logger.error(`Could not update gestionnaires for ${email} in database`);
+          logger.error(e);
+        }
+      }
+      stats.totalUpdated++;
```

Comment:
Est-ce que ça doit être augmenté s'il y a une erreur `could not update gestionnaires for email...` ? Cad  est-ce qu'il faut le laisser là ou le mettre juste après l'update ?

Reply:
En fait ici je veux affiche cette info même en dry run c'est pour ça que c'est là, mais je suis d'accord que c'e'st pas parfait, je change